src/modm/board/nucleo_g431rb/board.hpp

```diff
@@ -111,18 +112,22 @@ using Button = GpioInverted<GpioInputC13>;
 using LedD13 = D13;
 
 using Leds = SoftwareGpioPort< LedD13 >;
-
+/// @}
 
 namespace stlink
 {
+/// @ingroup modm_board_nucleo_g474re
```

Comment:
There are very few "wrong" `ingroup` commands using a small script:

```
ERROR: Probably a bad doxygen group ('driver_mcp23x17') in file 'src/modm/driver/gpio/mcp23_transport.hpp'.
ERROR: Probably a bad doxygen group ('driver_mcp23x17') in file 'src/modm/driver/gpio/mcp23_transport.hpp'.
ERROR: Probably a bad doxygen group ('platform_uart') in file 'src/modm/driver/io/terminal.hpp'.
ERROR: Probably a bad doxygen group ('platform_uart') in file 'src/modm/driver/io/terminal.hpp'.
ERROR: Probably a bad doxygen group ('driver_ea_dog') in file 'src/modm/driver/display/st7036.hpp'.
ERROR: Probably a bad doxygen group ('driver_ea_dog') in file 'src/modm/driver/display/st7036.hpp'.
ERROR: Probably a bad doxygen group ('driver_ea_dog') in file 'src/modm/driver/display/st7565_defines.hpp'.
ERROR: Probably a bad doxygen group ('driver_ea_dog') in file 'src/modm/driver/display/st7565_defines.hpp'.
ERROR: Probably a bad doxygen group ('driver_ea_dog') in file 'src/modm/driver/display/st7565.hpp'.
ERROR: Probably a bad doxygen group ('driver_ea_dog') in file 'src/modm/driver/display/st7565.hpp'.
ERROR: Probably a bad doxygen group ('driver_i2c_eeprom') in file 'src/modm/driver/storage/cat24aa.hpp'.
ERROR: Probably a bad doxygen group ('driver_i2c_eeprom') in file 'src/modm/driver/storage/cat24aa.hpp'.
ERROR: Probably a bad doxygen group ('board_black_pill_f401') in file 'src/modm/board/black_pill_f411/board.hpp'.
ERROR: Probably a bad doxygen group ('board_nucleo_g474re') in file 'src/modm/board/nucleo_g431rb/board.hpp'.
ERROR: Probably a bad doxygen group ('board_nucleo_g474re') in file 'src/modm/board/nucleo_g431rb/board.hpp'.
ERROR: Probably a bad doxygen group ('board_nucleo_l552ze_q') in file 'src/modm/board/nucleo_l552-ze-q/board.hpp'.
ERROR: Probably a bad doxygen group ('board_nucleo_l552ze_q') in file 'src/modm/board/nucleo_l552-ze-q/board.hpp'.
ERROR: Probably a bad doxygen group ('architecture_1_wire') in file 'src/modm/architecture/interface/one_wire.hpp'.
ERROR: Probably a bad doxygen group ('math_matrix') in file 'src/modm/math/lu_decomposition.hpp'.

Found 19 suspicious doxygen groups in 727 ingroup commands.
```

Maybe we could add the script to the CI...? (But I don't know what is the best way to handle the false positives.)

```python
#!/usr/bin/env python3

import re
import glob
from pathlib import Path

files = list(Path("src/").glob("**/*.hpp")) + list(Path("src/").glob("**/*.cpp"))
regex = r"[\\@]ingroup\s+modm_(?P<groupname>[0-9a-z_]+)"
error_counter = 0
ingroup_counter = 0

for f in files:
    for l in open(f, "r"):
        match = re.search(regex, l)
        if match:
            ingroup_counter += 1
            groupname = match.group("groupname")
            for s in groupname.split("_"):
                if s.rstrip("x") not in str(f):
                    print(f"ERROR: Probably a bad doxygen group ('{groupname}') in file '{str(f)}'.")
                    error_counter += 1

print("")
print(f"Found {error_counter} suspicious doxygen groups in {ingroup_counter} ingroup commands.")
exit(error_counter)
```

Reply:
I thought about writing a script that runs an lbuild simulation to get the build log and then checks that the ingroup maps onto the module that generated the file. That would give fewer false positives. But I'm not motivated to do that now, this was supposed to just be a "quick" fix that escalated as always 😜